contracts/MetaDataService.sol

```diff
@@ -0,0 +1,15 @@
+pragma solidity >=0.8.4;
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract MetaDataService {
+    using Strings for uint256;
+    string private _uri;
+
+    constructor(string memory _metaDataUri) {
+        _uri = _metaDataUri;
+    }
+
+    function uri(uint256 tokenId) public view returns (string memory) {
```

Comment:
This can just return the `metadataURI`, because the URI can contain `{id}` which the client replaces with the ID.